app/models/lecture.rb

```diff
@@ -1,3 +1,16 @@
 class Lecture < ApplicationRecord
   belongs_to :question_set
+  
+  #compute out of all the responses, how many corresponded to each answer
+  #returns hash keyed by answer id with value for % responses with that answer id
+  def self.compute_percentages (q)
+    @responses = Response.where(question: q)
+    @answers = Answer.where(question: q)
+    @total = @responses.length
+    hash = Hash.new(0)
```

Comment:
Also I think the argument to `Hash.new` is a default value if nothing has been inserted with the key. So if you are giving it `0` I don't think there is a need for `@answers.each{|key| hash[key.id] = 0}`